packages/connect-node/src/grpc-http-transport.ts

```diff
@@ -0,0 +1,408 @@
+            );
+            validateResponse(
+              useBinaryFormat,
+              response.statusCode,
+              responseHeaders
+            );
+
+            const messageResult = await readEnvelope(response);
+
+            let trailer = mapResponseTrailers(response.trailers);
+            if (receivedTrailers(trailer)) {
+              validateGrpcStatus(trailer);
+            }
+
+            if (messageResult.done) {
+              throw "premature eof";
+            }
+
+            const eofResult = await readEnvelope(response);
+            trailer = mapResponseTrailers(response.trailers);
+            if (receivedTrailers(trailer)) {
+              validateGrpcStatus(trailer);
+            }
+
+            if (!eofResult.done) {
+              throw "extraneous data";
+            }
+
+            return <UnaryResponse<O>>{
+              stream: false,
+              service,
+              method,
+              header: responseHeaders,
+              message: parse(messageResult.value.data),
+              trailer,
+            };
+          },
+          options.interceptors
+        );
+      } catch (e) {
+        throw connectErrorFromNodeReason(e);
+      }
+    },
+    async stream<
+      I extends Message<I> = AnyMessage,
+      O extends Message<O> = AnyMessage
+    >(
+      service: ServiceType,
+      method: MethodInfo<I, O>,
+      signal: AbortSignal | undefined,
+      timeoutMs: number | undefined,
+      header: HeadersInit | undefined
+    ): Promise<StreamingConn<I, O>> {
+      const { normalize, serialize, parse } = createClientMethodSerializers(
+        method,
+        useBinaryFormat,
+        options.jsonOptions,
+        options.binaryOptions
+      );
+
+      return runStreaming<I, O>(
+        {
+          stream: true,
+          service,
+          method,
+          url: createMethodUrl(options.baseUrl, service, method).toString(),
+          init: {
+            method: "POST",
+            redirect: "error",
+            mode: "cors",
+          },
+          signal: signal ?? new AbortController().signal,
+          header: grpcCreateRequestHeader(useBinaryFormat, timeoutMs, header),
+        },
+        (req: StreamingRequest<I, O>) => {
+          try {
+            const endpoint = new URL(req.url);
+            const nodeRequestFn = nodeRequest(endpoint.protocol);
+            const stream = nodeRequestFn(req.url, {
+              headers: webHeaderToNodeHeaders(req.header),
+              method: "POST",
+              path: endpoint.pathname,
+              signal: req.signal,
+              ...options.httpOptions,
+            });
+
+            const responsePromise = new Promise<http.IncomingMessage>(
+              (resolve, reject) => {
+                stream.on("response", (res) => {
+                  resolve(res);
+                });
+                stream.on("error", reject);
+              }
+            );
+            const responseTrailer = defer<Headers>();
```

Comment:
Can you switch this to a promise from `readHttp1ResponseTrailer()` as well? The diff between grpc-http-transport.ts and grpc-http2-transport.ts should be as small as possible. (Don't put extra work on it now, but please make sure we handle trailers exactly the same).